ampd/src/stellar/rpc_client.rs

```diff
@@ -28,13 +28,22 @@ const STATUS_SUCCESS: &str = "SUCCESS";
 
 impl From<(Hash, GetTransactionResponse)> for TxResponse {
     fn from((transaction_hash, response): (Hash, GetTransactionResponse)) -> Self {
-        let contract_events = response
-            .result_meta
-            .and_then(|meta| match meta {
-                TransactionMeta::V3(data) => data.soroban_meta.map(|meta| meta.events),
-                _ => None,
-            })
-            .unwrap_or_default();
+        // Protocol 23 (CAP-0067): Extract contract events from the unified events structure
+        // contract_events is Vec<Vec<ContractEvent>> (per operation), so we flatten it
+        let events_vec: Vec<ContractEvent> = response
```

Comment:
```suggestion
        let events: Vec<ContractEvent> = response
```
avoid type suffix in variable names